src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2654,6 +2657,102 @@ private Process StartWithShellExecute(ProcessStartInfo startInfo)
+    {
+        private const string StartProcessCommandName = "Start-Process";
+        private const string FilePathParameterName = "FilePath";
+        private const string GetCommandCommandName = "Get-Command";
+        private const string CommandNameParameterName = "Name";
+        private const string CommandTypeParameterName = "CommandType";
+
+        /// <summary>
+        /// Returns completion results for verb parameter.
+        /// </summary>
+        /// <param name="commandName">The command name.</param>
+        /// <param name="parameterName">The parameter name.</param>
+        /// <param name="wordToComplete">The word to complete.</param>
+        /// <param name="commandAst">The command AST.</param>
+        /// <param name="fakeBoundParameters">The fake bound parameters.</param>
+        /// <returns>List of Completion Results.</returns>
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters)
+        {
+            // -Verb is not supported on non-Windows platforms as well as Windows headless SKUs
+            if (!Platform.IsWindowsDesktop)
+            {
+                yield break;
+            }
+
+            // Completion: Start-Process -FilePath <path> -Verb <wordToComplete>
+            if (commandName.Equals(StartProcessCommandName, StringComparison.OrdinalIgnoreCase)
+                && fakeBoundParameters.Contains(FilePathParameterName))
+            {
+                string filePath = fakeBoundParameters[FilePathParameterName].ToString();
+
+                // Complete file verbs if extension exists
+                if (Path.HasExtension(filePath))
+                {
+                    foreach (string verb in CompleteFileVerbs(filePath, wordToComplete))
+                    {
+                        yield return new CompletionResult(verb);
+                    }
+
+                    yield break;
+                }
+
+                // Otherwise check if command is an Application to resolve executable full path with extension
+                // e.g if powershell was given, resolve to powershell.exe to get verbs
+                using (var ps = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace))
+                {
+                    var commandInfo = new CmdletInfo(GetCommandCommandName, typeof(GetCommandCommand));
+
+                    ps.AddCommand(commandInfo);
+                    ps.AddParameter(CommandNameParameterName, filePath);
+                    ps.AddParameter(CommandTypeParameterName, CommandTypes.Application);
+
+                    Collection<CommandInfo> commands = ps.Invoke<CommandInfo>();
+
+                    // PATHEXT gives us executable command extensions in order so we can complete verbs from first extension
```

Comment:
This comment does not reflect the most important thing - that both Start-Process and Get-Command follow the behavior of the OS, i.e. they select the first found application (based on PATHEXT) and _this is the same value for both commands_.

Reply:
Fixed